app/src/main/java/org/fossasia/openevent/general/search/SearchFilterFragment.kt

```diff
@@ -1,39 +1,44 @@
 package org.fossasia.openevent.general.search
 
-import android.graphics.drawable.Drawable
 import android.os.Bundle
 import android.view.LayoutInflater
 import android.view.Menu
 import android.view.MenuInflater
 import android.view.MenuItem
 import android.view.View
 import android.view.ViewGroup
-import androidx.appcompat.app.AppCompatActivity
 import androidx.fragment.app.Fragment
 import androidx.navigation.Navigation
 import androidx.navigation.fragment.navArgs
 import kotlinx.android.synthetic.main.fragment_search_filter.view.freeStuffCheckBox
+import kotlinx.android.synthetic.main.fragment_search_filter.view.tvSelectCategory
+import kotlinx.android.synthetic.main.fragment_search_filter.view.tvSelectDate
+import kotlinx.android.synthetic.main.fragment_search_filter.view.tvSelectLocation
 import org.fossasia.openevent.general.R
 import org.fossasia.openevent.general.utils.Utils.getAnimFade
 import org.fossasia.openevent.general.utils.Utils.setToolbar
+import org.koin.androidx.viewmodel.ext.android.viewModel
 
 class SearchFilterFragment : Fragment() {
 
     private lateinit var rootView: View
     private var isFreeStuffChecked = false
+    private var selectedTime: String? = null
+    private var selectedLocation: String? = null
+    private var selectedCategory: String? = null
```

Comment:
why not `lateinit` ?

Reply:
I read somewhere along issues where areebJamal said "use lateinit as last resort". No other specific reason.